events.go

```diff
@@ -0,0 +1,196 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"net/url"
+	"reflect"
+	"strings"
+	"time"
+	"unicode"
+
+	"github.com/openfaas-incubator/connector-sdk/types"
+	"github.com/pkg/errors"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/event"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/soap"
+	vtypes "github.com/vmware/govmomi/vim25/types"
+)
+
+// OutboundEvent is the JSON object sent to subscribed functions
+// If the ManagedObjectReference for an event cannot be retrieved, it will be nil and thus not marshaled into the JSON OutboundEvent
+// It's the receivers responsibility to check whether managedObjectReference key is present in the JSON message payload
+type OutboundEvent struct {
```

Comment:
I'm not sure that having multiple files makes sense when they are all in the same package.

If you want to split them out then consider also breaking out a package / sub-folder.